### ISSUE TITLE
Unicorn.aspx indicates Transparent Sync is enabled on configs but TS is not working and does not display in Content Editor warning

### ISSUE THREAD

**Issue by strice** [SitecoreUnicorn/Unicorn#420]
**Do you want to request a *feature* or report a *bug*?**

Bug report

**What is the current behavior?**

We are working with a Sitecore 10.1.2 instance where we have Transparent Sync enabled on multiple Unicorn configurations. On `Unicorn.aspx` the relevant configurations indicate they are using TS.

![image](https://user-images.githubusercontent.com/1474105/142284115-5663264e-4c9c-4f59-a5c6-73bdebc62c61.png)

However, Transparent Sync is not taking place when items are changed or serialized files are updated. In addition, the Content Editor warning does not indicate Transparent Sync is enabled when viewing an item that falls under one the configurations that should have TS enabled.

![image](https://user-images.githubusercontent.com/1474105/142285599-c29c0116-3174-4341-ba85-0061514716dc.png)

**If the current behavior is a bug, please provide the steps to reproduce.**

Here is the `<unicorn />` section of `showconfig.aspx`. Please let me know what else would be appropriate to help troubleshoot/reproduce.

[unicorn_snippet.zip](https://github.com/SitecoreUnicorn/Unicorn/files/7558078/unicorn_snippet.zip)

I could not identify anything related to this issue in the Sitecore logs.

**What is the expected behavior?**

Transparent Sync should be functioning on those configurations and Content Editor warning should indicate it as well.

![image](https://user-images.githubusercontent.com/1474105/142285641-51255695-da86-42c4-ab5c-27346a954ed1.png)

**Please mention your Sitecore version and Unicorn version.**

Sitecore 10.1.2
Unicorn 4.1.5

**Comment by cassidydotdk** [comment]
Hi.

What role is your CM server in?   It would be `StandAlone` or `ContentManagement`.

Can you include your Unicorn.DataProvider.config, please?   You would need to have the 10.1 version enabled for TS to kick in.

**Comment by cassidydotdk** [comment]
Assuming the DataProvider was missing.

**Comment by strice** [comment]
Hi @cassidydotdk - 

Apologies for the delayed response (holidays and all). 

The role:
`<add key="role:define" value="ContentManagement, Indexing" />`

Unicorn.DataProvider.config:
[Unicorn.DataProvider.config.txt](https://github.com/SitecoreUnicorn/Unicorn/files/7627580/Unicorn.DataProvider.config.txt)

In reviewing this, I did notice some additional behavior -

Returning to this after a few weeks, I observed that the content editor warning on the templates DID display the Transparent Sync message as expected - 
![before touching](https://user-images.githubusercontent.com/1474105/144098993-00bfeeb3-b17b-45fa-9eb9-43d4971385ef.PNG)

I then changed the field type on that field as a test and I observed the Transparent Sync warning NOT displaying any longer after saving that item - 
![after touching](https://user-images.githubusercontent.com/1474105/144099238-c15a6011-071e-497c-9f48-920e6f2b9585.PNG)
In addition, the changes were NOT reserialized automatically. Oddly enough, the parent items still displayed the Transparent Sync warning (although they exhibit the same behavior if changes are made directly to them). 

After resyncing/reserializing that item continues to not display the Transparent Sync warning.
![after resync](https://user-images.githubusercontent.com/1474105/144099539-d2db2fc2-530c-4b37-aea7-0ae577b9042c.PNG)

Even after an app restart, the TS warning does not reappear. I am not sure what is causing the TS warning to appear/reappear. At no point did the items automatically reserialize after changing them.
